File: backend/src/main/java/com/plugga/pluggarestapi/service/UserService.java
package com.plugga.pluggarestapi.service;

import java.util.List;

import com.plugga.pluggarestapi.entity.User;

public interface UserService {

    public List<User> findAll();
    
    public User findById(int id);
    
    public void save(User user);
    
    public void deleteById(int id);
}

File: backend/src/main/kotlin/com/plugga/backend/service/UserService.kt
package com.plugga.backend.service

import com.plugga.backend.entity.User
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface UserService {

    fun findAll(pageable: Pageable): Page<User>

    fun findById(id: Int): User?

    fun saveUser(user: User): User?

    fun deleteById(id: Int)
}
